src/kimmdy/topology/topology.py

```diff
@@ -856,6 +859,7 @@ def reindex_atomnrs(self) -> dict[str, dict[str, str]]:
         update_map_all = {}
         for id, moleculetype in self.moleculetypes.items():
             update_map_all[id] = moleculetype.reindex_atomnrs()
```

Comment:
I think we can simply this with the knowledge that only atoms in the `Reactive` moleculetype are meant to be modified, but that's something for a follow-up